src/list.cpp

```diff
@@ -254,39 +401,66 @@ util::Optional<Mixed> List::min(size_t column)
     return as_results().min(column);
 }
 
-util::Optional<Mixed> List::sum(size_t column)
+Mixed List::sum(size_t column)
 {
-    return as_results().sum(column);
+    // Results::sum() returns none only for Empty results
```

Comment:
In Mode::Empty it doesn't know what type of value to return since it gets the type from the table, and it can't get it from the ObjectSchema because it looks that up from the Table name. This is all a pre-existing design issue that should probably be fixed at some point.